packages/core/src/util/requireHook.js

```diff
@@ -54,6 +54,25 @@ exports.init = function (config) {
  * @param {string} moduleName
  */
 function patchedModuleLoad(moduleName) {
+  let module = moduleName;
+
+  // CASE: when using ESM, the Node runtime passes a full path to Module._load
+  //       we try to grab the module name to being able to patch the target module
+  //       with our instrumentation
+  if (path.isAbsolute(moduleName) && moduleName.indexOf('.js') !== -1) {
+    try {
+      // node_modules/mysql/lib/index.js
+      module = moduleName.match(/node_modules\/(.*?(?=\/))/)[1];
```

Comment:
_nitpick:_ Looks like we are relying on the `[1]` access to throw a TypeError when there is no match and `match` returns `null`. Can we replace that with an actual null check on the result of `match` instead of using exceptions for flow control?

Reply:
Sure why not